ui/src/app/components/object-detail/object-detail.component.html

```diff
@@ -0,0 +1,522 @@
+                    #matSelectspDataType
+                    [formControl]="element.get('spDataType')"
+                    selected="element.get('spDataType').value"
+                    (selectionChange)="spTableEditSuggestionHandler(i, matSelectspDataType.value)"
+                  >
+                    <mat-option
+                      *ngFor="let t of typeMap[element.get('srcDataType').value]"
+                      [value]="t.T"
+                    >
+                      {{ t.T }}
+                    </mat-option>
+                  </mat-select>
+                </mat-form-field>
+                <p *ngIf="!isEditMode">{{ element.get('spDataType').value }}</p>
+              </div>
+            </td>
+          </ng-container>
+
+          <ng-container matColumnDef="spIsPk">
+            <th class="table_header" mat-header-cell *matHeaderCellDef>Pk</th>
+            <td mat-cell *matCellDef="let element">
+              <div *ngIf="element.get('spIsPk').value">
+                <svg
+                  width="14"
+                  height="8"
+                  viewBox="0 0 14 8"
+                  fill="none"
+                  xmlns="http://www.w3.org/2000/svg"
+                >
+                  <path
+                    fill-rule="evenodd"
+                    clip-rule="evenodd"
+                    d="M4 8C5.86384 8 7.42994 6.72523 7.87398 5H10V7H13V5H14V3H7.87398C7.42994 1.27477 5.86384 0 4 0C1.79086 0 0 1.79086 0 4C0 6.20914 1.79086 8 4 8ZM6 4C6 5.10457 5.10457 6 4 6C2.89543 6 2 5.10457 2 4C2 2.89543 2.89543 2 4 2C5.10457 2 6 2.89543 6 4Z"
+                    fill="black"
+                  />
+                </svg>
+              </div>
+            </td>
+          </ng-container>
+
+          <ng-container matColumnDef="spIsNotNull">
+            <th mat-header-cell class="table_header" *matHeaderCellDef>
+              <span *ngIf="isEditMode">Not Null</span><span *ngIf="!isEditMode">Nullable</span>
+            </th>
+            <td mat-cell *matCellDef="let element">
+              <div class="notNullContent">
+                <div *ngIf="isEditMode && element.get('spOrder').value !== ''">
+                  <mat-checkbox
+                    color="primary"
+                    [formControl]="element.get('spIsNotNull')"
+                  ></mat-checkbox>
+                </div>
+                <p *ngIf="!isEditMode">
+                  {{ element.get('spIsNotNull').value ? 'Not Null' : '' }}
+                </p>
+              </div>
+            </td>
+          </ng-container>
+
+          <ng-container matColumnDef="dropButton">
+            <th mat-header-cell class="table_header" *matHeaderCellDef></th>
+            <td mat-cell *matCellDef="let element">
+              <div
+                class="actions"
+                *ngIf="currentObject!.isSpannerNode && !(element.get('spOrder').value === '')"
+              >
+                <mat-icon [matMenuTriggerFor]="menu">more_vert</mat-icon>
```

Comment:
Actually this icon and the option to drop column should only come up when edit button is clicked, I think same is the flow in figma. We can discuss more on this if required.

Reply:
done